core/services/ardupilot_manager/main.py

```diff
@@ -28,13 +43,42 @@ def check_running_as_root() -> None:
         if os.geteuid() != 0:
             raise RuntimeError("ArduPilot manager needs to run with root privilege.")
 
-    @staticmethod
-    def start_navigator() -> None:
-        raise NotImplementedError("We only support ArduPilot running on serial devices.")
+    def start_navigator(self) -> None:
+        firmware = os.path.join(self.firmware_path, "ardusub")
+        if not os.path.isfile(firmware):
+            temporary_file = self.firmware_download.download(Vehicle.Sub, "Navigator")
+            assert temporary_file, "Failed to download navigator binary."
+            os.chmod(temporary_file, stat.S_IXOTH)
```

Comment:
I wish I could go back in time and hit whoever thought naming something `S_IXOTH` was a good idea.
Can you add a comment saying that it just adds execute permission?